adoc/scripts/vkconventions.py

```diff
@@ -0,0 +1,298 @@
+    'Win32',  # VkWin32SurfaceCreateInfoKHR
+))
+# A regex to match any of the SPECIAL_WORDS
+EXCEPTION_PATTERN = r'(?P<exception>{})'.format(
+    '|'.join('(%s)' % re.escape(w) for w in SPECIAL_WORDS))
+MAIN_RE = re.compile(
+    # the negative lookahead is to prevent the all-caps pattern from being too greedy.
+    r'({}|([0-9]+)|([A-Z][a-z]+)|([A-Z][A-Z]*(?![a-z])))'.format(EXCEPTION_PATTERN))
+
+
+class VulkanConventions(ConventionsBase):
+    @property
+    def null(self):
+        """Preferred spelling of NULL."""
+        return '`NULL`'
+
+    def formatExtension(self, name):
+        """Mark up an extension name as a link the spec."""
+        return '`apiext:{}`'.format(name)
+
+    @property
+    def struct_macro(self):
+        """Get the appropriate format macro for a structure.
+
+        Primarily affects generated valid usage statements.
+        """
+
+        return 'slink:'
+
+    @property
+    def constFlagBits(self):
+        """Returns True if static const flag bits should be generated, False if an enumerated type should be generated."""
+        return False
+
+    @property
+    def structtype_member_name(self):
+        """Return name of the structure type member"""
+        return 'sType'
+
+    @property
+    def nextpointer_member_name(self):
+        """Return name of the structure pointer chain member"""
+        return 'pNext'
+
+    @property
+    def valid_pointer_prefix(self):
+        """Return prefix to pointers which must themselves be valid"""
+        return 'valid'
+
+    def is_structure_type_member(self, paramtype, paramname):
+        """Determine if member type and name match the structure type member."""
+        return paramtype == 'VkStructureType' and paramname == self.structtype_member_name
+
+    def is_nextpointer_member(self, paramtype, paramname):
+        """Determine if member type and name match the next pointer chain member."""
+        return paramtype == 'void' and paramname == self.nextpointer_member_name
+
+    def generate_structure_type_from_name(self, structname):
+        """Generate a structure type name, like VK_STRUCTURE_TYPE_CREATE_INSTANCE_INFO"""
+
+        structure_type_parts = []
+        # Tokenize into "words"
+        for elem in MAIN_RE.findall(structname):
+            word = elem[0]
+            if word == 'Vk':
+                structure_type_parts.append('VK_STRUCTURE_TYPE')
+            else:
+                structure_type_parts.append(word.upper())
+        name = '_'.join(structure_type_parts)
+
+        # The simple-minded rules need modification for some structure names
+        subpats = [
+            [ r'_H_(26[45])_',              r'_H\1_' ],
+            [ r'_VULKAN_([0-9])([0-9])_',   r'_VULKAN_\1_\2_' ],
+            [ r'_VULKAN_SC_([0-9])([0-9])_',r'_VULKAN_SC_\1_\2_' ],
+            [ r'_DIRECT_FB_',               r'_DIRECTFB_' ],
+            [ r'_VULKAN_SC_10',             r'_VULKAN_SC_1_0' ],
```

Comment:
A lot of this is Vulkan specific, do we need this, or even this file?

Reply:
It depends on your definition of `need` :). This file is "required" because of some import. But indeed, they seem to be related to some Vulkan conversion, so in theory, we can just do some modification and remove it. 

But saying that, I think the maintenance will be easier if we just copy/paste everything without "un-necessary-to-get-it-working" modification.  I agree that we don't need it, but it doesn't seem to hurt. The reformated code seems OK.

To be honest, I don't want to learn the inner architecture of `reflow.py` and neither supports a fancy fork. Nobody have time for that :)

Reply:
Easy maintenance is always good to have :)

Reply:
The different APIs using these scripts have different conventions and they are capture in the APIConventions interface. So if you needed to, you could create syclconventions.py with the SYCL-specific things and make apiconventions.py import that, instead of vkconventions.py. However, most of the conventions are used by other scripts which you have no current use for in SYCL, so you may not even need that much - you could try replacing the code in apiconventions.py with just

> from spec_tools.conventions import ConventionsBase as APIConventions

and if that works as desired, remove vkconventions.py.

Reply:
Tried and got 
```
TypeError: Can't instantiate abstract class ConventionsBase with abstract methods api_name, api_prefix, extension_file_path, file_suffix, generate_structure_type_from_name, nextpoin
ter_member_name, null, structtype_member_name, xml_api_name
```

And indeed, you were "too clean", and used `abc.abstractmethod` on those methods. IMO we can merge as is. In the future, if we need a new `syclConvention` we can implement those methods (and maybe refactor the `apiconventions.py` to get a new ENV variable or something around those line)